data_preparation/process_all_scan.py

```diff
@@ -0,0 +1,243 @@
+import json
+import os
+from argparse import ArgumentParser
+from functools import wraps
+from typing import Dict, Tuple
+
+import mmengine
+import numpy as np
+import torch
+from pytorch3d.transforms import euler_angles_to_matrix
+from tqdm import tqdm
+from utils.data_utils import read_annotation_pickle
+from utils.mp3d_process import process_mp3d
+from utils.pcd_utils import is_inside_box
+from utils.scannet_process import process_scannet
+from utils.trscan_process import process_3rscan
+
+es_anno = {}
+
+
+def mmengine_track_func(func):
```

Comment:
Is this function necessary?

Reply:
It is necessary if using the 'mmengine'  to run multi-process multi-parameter functions.